lib/internal/modules/cjs/loader.js

```diff
@@ -1004,8 +1004,13 @@ Module._resolveFilename = function(request, parent, isMain, options) {
     paths = Module._resolveLookupPaths(request, parent);
   }
 
-  if (parent && parent.filename) {
-    const filename = trySelf(parent.filename, isMain, request);
+  if (parent && (
+    parent.id === 'internal/preload' ||
+    parent.id === '<repl>' ||
+    parent.filename
+  )) {
+    const base = parent.filename || process.cwd() + path.sep;
```

Comment:
`process.cwd()` is causing `test/parallel/test-cwd-enoent-preload.js` to fail with error:

```bash
=== release test-cwd-enoent-preload ===                                       
Path: parallel/test-cwd-enoent-preload
internal/bootstrap/switches/does_own_process_state.js:129
    cachedCwd = rawMethods.cwd();
                           ^

Error: ENOENT: no such file or directory, uv_cwd
    at process.wrappedCwd [as cwd] (internal/bootstrap/switches/does_own_process_state.js:129:28)
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:1012:45)
    at Function.Module._load (internal/modules/cjs/loader.js:908:27)
    at Module.require (internal/modules/cjs/loader.js:1099:19)
    at Module._preloadModules (internal/modules/cjs/loader.js:1355:12)
    at loadPreloadModules (internal/bootstrap/pre_execution.js:448:5)
    at prepareMainThreadExecution (internal/bootstrap/pre_execution.js:71:3)
    at internal/main/eval_string.js:14:1 {
  errno: -2,
  code: 'ENOENT',
  syscall: 'uv_cwd'
}
assert.js:102
  throw new AssertionError(obj);
  ^

```

Reply:
Is this for the `parent.id === 'internal/preload'` case? Why is that necessary for the repl support?

Reply:
It is not necessary for the repl support but It is for preloading a module using `-r` from the command line. Manual testing works, for what it's worth. Could the use of [`chdir`](https://github.com/nodejs/node/blob/a022d387ea89a3bc5e7435faf2ae7030e0646af1/test/parallel/test-cwd-enoent-preload.js#L19) cause this? I don't know though. The error is complaining about the fact that  `cwd` does not exist.

Reply:
Or maybe [`tmpdir.refresh()`](https://github.com/nodejs/node/blob/a022d387ea89a3bc5e7435faf2ae7030e0646af1/test/parallel/test-cwd-enoent-preload.js#L17)

Reply:
Ah I see ok in that case perhaps just wrapping the `process.cwd()` in a try-catch can work around the issue? Effectively defaulting to the old behaviour in the catch case.